vendor/github.com/minio/minio-go/api-get.go

```diff
@@ -0,0 +1,300 @@
+package minio
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+)
+
+// GetBucketACL get the permissions on an existing bucket.
+//
+// Returned values are:
+//
+//  private - owner gets full access.
+//  public-read - owner gets full access, others get read access.
+//  public-read-write - owner gets full access, others get full access too.
+//  authenticated-read - owner gets full access, authenticated users get read access.
+func (a Client) GetBucketACL(bucketName string) (BucketACL, error) {
+	if err := isValidBucketName(bucketName); err != nil {
+		return "", err
+	}
+
+	// Set acl query.
+	urlValues := make(url.Values)
+	urlValues.Set("acl", "")
+
+	// Instantiate a new request.
+	req, err := a.newRequest("GET", requestMetadata{
+		bucketName:  bucketName,
+		queryValues: urlValues,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// Initiate the request.
+	resp, err := a.httpClient.Do(req)
+	defer closeResponse(resp)
+	if err != nil {
+		return "", err
+	}
+	if resp != nil {
+		if resp.StatusCode != http.StatusOK {
+			return "", BodyToErrorResponse(resp.Body)
+		}
+	}
+
+	// Decode access control policy.
+	policy := accessControlPolicy{}
+	err = xmlDecoder(resp.Body, &policy)
+	if err != nil {
+		return "", err
+	}
+
+	// If Google private bucket policy doesn't have any Grant list.
```

Comment:
> I would like more elaborate comment here. Any time we deviate from S3 API and do vendor specific checks, it is important to give full explanation on why we need to handle such exceptions. New contributors wont have any context otherwise.

Okay i will add.